src/components/organisms/Header/index.tsx

```diff
@@ -0,0 +1,193 @@
+  const [text, setText] = useState("");
+  const [suggestions, setSuggestions] = useState([]);
+  useEffect(() => {
+    api.get("/").then((response) => setData(response.data));
+  }, []);
+  const onChangeHandler = (text: string) => {
+    let matches = [];
+    if (text.length > 0) {
+      matches = data.filter((bookData: any) => {
+        const regex = new RegExp(`${text}`, "gi");
+        return bookData.name.match(regex);
+      });
+    }
+    setSuggestions(matches);
+    setText(text);
+  };
+
+  const handle = () => {
+    isOpen ? setIsOpen(false) : setIsOpen(true);
+    if (underLine !== "3px") {
+      setUnderLine("3px");
+
+      setIcon(up);
+    } else {
+      setUnderLine("0px");
+      setIcon(down);
+    }
+  };
+  const handleSearch = () => {
+    isSearchClick ? setIsSearchClick(false) : setIsSearchClick(true);
+  };
+  const handleProfile = () => {
+    isOpenProfile ? setIsOpenProfile(false) : setIsOpenProfile(true);
+    if (isOpenProfile === true) {
+      setProfileIcon(down);
+    } else {
+      setProfileIcon(up);
+    }
+  };
+  const onSuggestHandler = (text) => {
+    setText(text);
+    setSuggestions([]);
+  };
+
+  const { user } = useAuth0();
+
+  return (
+    <>
+      <AppBar elevation={0}>
+        <Toolbar>
+          <div>
+            <div>
+              <img src={logo} alt="logo" />
+              <div
+                onClick={() => {
+                  console.log("search clicked");
+                  handleSearch();
+                }}
+                style={{ cursor: "pointer" }}
+              >
+                <img src={searchicon} alt="search" />
+                <div>
+                  <div>
+                    <Modal
+                      open={isSearchClick}
+                      DropComp={() => {
+                        return (
+                          <>
+                            {suggestions &&
+                              suggestions.map((suggestion: any, i) => {
+                                return (
+                                  <>
+                                    <div
+                                      className={classes.suggestion}
+                                      key={i}
+                                      onClick={() => {
+                                        onSuggestHandler(suggestion.name);
+                                        navigate(
+                                          `/bookdetailviewpage/${suggestion.id}`
+                                        );
+                                      }}
+                                    >
+                                      {suggestion.name +
+                                        " by " +
+                                        suggestion.author}
+                                    </div>
+                                  </>
+                                );
+                              })}
+                          </>
+                        );
```

Comment:
can you refactor this into a separate component something like SearchResults component